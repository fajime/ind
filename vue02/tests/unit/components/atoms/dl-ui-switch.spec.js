import { mount } from '@vue/test-utils';
import randomIdMixin from '../../../../src/mixins/randomId';
import DlUiSwitch from '../../../../src/components/atoms/dl-ui-switch/index.vue';

describe('dl-ui-switch', () => {
  randomIdMixin.methods.createRandomId = () => 'random-id';
  const wrapper = mount(DlUiSwitch);

  it('Component Name', () => expect(DlUiSwitch.name).toBe('dl-ui-switch'));

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has input', () => {
    expect(wrapper.find('input').exists()).toBe(true);
  });

  describe('methods', () => {

    it('raiseEvent', async () => {
      const wrapperTest = mount(DlUiSwitch,
        {
          props : {
            disabled   : false,
            modelValue : false
          }
        });
      wrapperTest.find('input').trigger('input');
      await wrapperTest.vm.$nextTick();
      expect(wrapperTest.emitted()).toHaveProperty('update:modelValue');
    });
  });
});

