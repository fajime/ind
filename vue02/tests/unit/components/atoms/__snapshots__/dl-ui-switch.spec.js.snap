// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dl-ui-switch renders correctly 1`] = `
<div
  class="dl-ui-switch"
>
  <!--v-if-->
  <span
    class="dl-ui-switch__span"
  >
    <input
      id="random-id"
      type="checkbox"
    />
  </span>
</div>
`;
